spark-bigquery-dsv2/spark-bigquery-dsv2-common/src/main/java/com/google/cloud/spark/bigquery/v2/context/BigQueryDataSourceReaderContext.java

```diff
@@ -316,9 +316,10 @@ public Filter[] pushFilters(Filter[] filters) {
         unhandledFilters.add(filter);
       }
     }
-
-    pushedFilters = handledFilters.stream().toArray(Filter[]::new);
-    return unhandledFilters.stream().toArray(Filter[]::new);
+    // We tell Spark that all filters were unhandled, in order to trigger bloom filter, DPP if
+    // needed
+    pushedFilters = filters;
```

Comment:
This line ignores the results of `isTopLevelFieldHandled()`

Reply:
Should I remove the calculation of handled, unhandled filters? Since we will no longer use it.

Reply:
We need to know which filters to push down to the ReadSession creation API